packages/astro/src/runtime/client/hmr.ts

```diff
@@ -38,3 +69,38 @@ if (import.meta.hot) {
 		await updateAll(event.updates);
 	});
 }
+
+const sheetsMap = new Map();
+
+function updateStyle(id: string, content: string): void {
+	let style = sheetsMap.get(id);
+	if (style && !(style instanceof HTMLStyleElement)) {
+		removeStyle(id);
+		style = undefined;
+	}
+
+	if (!style) {
+		style = document.createElement('style');
+		style.setAttribute('type', 'text/css');
+		style.innerHTML = content;
+		document.head.appendChild(style);
+	} else {
+		style.innerHTML = content;
+	}
+	sheetsMap.set(id, style);
```

Comment:
Keeps track of style blocks by file ID to make sure they are replaced on the next HMR update (instead of duplicating the styles)